component/component.go

```diff
@@ -29,6 +29,11 @@ type Component interface {
 	// Start() then the collector startup will be aborted.
 	// If this is an exporter component it may prepare for exporting
 	// by connecting to the endpoint.
+	//
+	// The context passed to this method must be used only in the scope of Start, and
+	// should not be stored or reused in any async work that the component executes
+	// after the start. It is important because the context may be cancelled by the
+	// service at any moment between Start and Shutdown.
 	Start(ctx context.Context, host Host) error
 
 	// Shutdown is invoked during service shutdown.
```

Comment:
Can we also add the following clarification to Shutdown()?
>If there are any background operations running by the component they must be aborted as soon as possible.
>Remember that if you started any long-running background operation from the Start() method that operation
>must be also cancelled.

Reply:
done.